forte/data/data_store.py

```diff
@@ -425,7 +434,52 @@ def _is_annotation(self, type_name: str) -> bool:
         entry_class = get_class(type_name)
         return issubclass(entry_class, (Annotation, AudioAnnotation))
 
-    def add_annotation_raw(self, type_name: str, begin: int, end: int) -> int:
+    def _add_entry_raw(
+        self,
+        entry_type: Union[Type[Annotation], Type[Link], Type[Group]],
+        type_name: str,
+        entry: List[Any],
+    ):
+        """
+        This function add raw entry in DataStore object
+        based on corresponding type name
+         and sort them based on entry type.
+
+        Args:
+            entry_type: entry's type which decides the sorting of entry.
+            type_name: the fully qualified type name of the new Annotation.
+            entry: raw entry data in the list format.
+
+        Raises:
+            ValueError: raised when the entry type's sorting function is not
+            implemented.
+
+        Returns:
+            ``tid`` of the entry.
+        """
+        if entry_type == Annotation:
+            sorting_fn = lambda s: (
+                s[constants.BEGIN_INDEX],
+                s[constants.BEGIN_INDEX],
+            )
+        elif entry_type in [Link, Group]:
+            sorting_fn = lambda s: (s[constants.ENTRY_TYPE_INDEX])
```

Comment:
Actually, I don't think we need sorting for `Link` or `Group`:
1. They are sorted based on the insertion order
2. We don't want to maintain sorted list here
3. Each entry (link or group) are adding to their own list, so the sorting here is not needed (they will be all the same for the same element list)